### PR TITLE
Adding Config Directory

### PR DESCRIPTION
**Pull request by assaftal391**
Please note that there is import problem while running main file.

### CODE REVIEW

**Comment by assaftal391** [comment]
I updated to Yaml files for configuration. Now the app is not crashing, however I tried to get the home page and got 404 error. Need to understand why.

**Comment by assaftal391** [comment]
Testing env running on localhost port 5001 and Dev env on port 5000.

File: soccer_application/__init__.py [hunk 1]
```diff
@@ -3,17 +3,35 @@
...
-login_manager = LoginManager(app)
+def create_app(config_name):
+    app = Flask(__name__)
+    config_dir = os.path.join(os.path.dirname(__file__), 'config')
+    if config_name == 'dev':
+        config_file = os.path.join(config_dir, 'dev.yaml')
+    elif config_name == 'testing':
+       config_file = os.path.join(config_dir, 'testing.yaml')
+    else:
+        raise ValueError(f"Invalid configuration name {config_name}")
```
Comment (YossiHananya) [inline soccer_application/__init__.py:21]: 1. Enum is preferable instead of strings comparison - let's change it to enum. 
2. The config initialization conditions should be part of the Config class, also the "create_app" method should stay as clear as we can. Therefore we should move those lines to the __init__ method of the config path or to a new class method ("from_path" or "from_type"), class method is preferable in my opinion.

File: soccer_application/forms.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def validate_username(self, username):
             raise ValidationError("Username already exists")
 
     def validate_email(self, email):        
-        if User.query.filter_by(email=email.data).first():
+        if User.query.filter_by(email=email.data).first() or User.query.filter_by(email==email.data).first().lower()==email.data.lower():
```
Comment (YossiHananya) [inline soccer_application/forms.py:50]: Submit this change in another branch.

File: soccer_application/config/config.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import yaml
+
+class Config:
+
+    def __init__(self,config_file):
+        with open(config_file,'r') as f:
+            self.config_data=yaml.safe_load(f)
+
```
Comment (YossiHananya) [inline soccer_application/config/config.py:8]: Nice to have - add __getitem__ method to the class so we can use the instance as a dict.

File: main.py [hunk 4]
```diff
@@ -1,29 +1,7 @@
...
-def main(port, debug):
-    app.run(host='localhost',port=port, debug=debug)
+from soccer_application import create_app
 
 
 if __name__=="__main__":
-    args = vars(menu())
-    main(**args)
+    app=create_app('dev')
+    app.run(host='localhost',port=app.config.get('PORT'))
```
Comment (YossiHananya) [inline main.py:7]: What about the "Debug" value in the app.run()?

File: soccer_application/api.py [hunk 5]
```diff
@@ -1,10 +1,11 @@
 import json
 from flask import jsonify, request, render_template, url_for, flash, redirect
 from flask_login import login_user, logout_user, current_user 
-from soccer_application import app, db, bcrypt
+from soccer_application import db, bcrypt, create_app
 from soccer_application.forms import RegistrationForm,LoginForm
 from soccer_application.models import Player, User
 
+app=create_app('testing')
```
Comment (YossiHananya) [inline soccer_application/api.py:8]: Why did you add this line?

File: soccer_application/__init__.py (hunk 1 above)
Comment (assaftal391) [reply soccer_application/__init__.py]: Changed as requested

File: soccer_application/forms.py (hunk 2 above)
Comment (assaftal391) [reply soccer_application/forms.py]: Deleted

File: soccer_application/config/config.py (hunk 3 above)
Comment (assaftal391) [reply soccer_application/config/config.py:10]: Didn't really understand.

File: main.py (hunk 4 above)
Comment (assaftal391) [reply main.py:7]: Inside the configuration files

File: soccer_application/api.py (hunk 5 above)
Comment (assaftal391) [reply soccer_application/api.py]: Because this function takes an argument "config_name"

Comment (assaftal391) [reply soccer_application/api.py]: Fixed this issue with addin "current_app" from flask and changing accordingly the init.py file

File: soccer_application/config/config.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    def get(self,key):
+        return self.config_data.get(key)
+    
+    @classmethod
+    def from_type(cls, config_value=ConfigName.DEVELOPMENT.value):
```
Comment (YossiHananya) [inline soccer_application/config/config.py:15]: Change the "config_value" parameter to an enum, it help us write the function without duplications 

Instead of ConfigName.Development.value
Change to ConfigName.Development

And while calling the function use the enum instead of it's values

File: main.py [hunk 7]
```diff
@@ -1,29 +1,7 @@
...
-
-def main(port, debug):
-    app.run(host='localhost',port=port, debug=debug)
+from soccer_application import create_app
 
 
 if __name__=="__main__":
-    args = vars(menu())
-    main(**args)
+    app=create_app('testing')
```
Comment (YossiHananya) [inline main.py:6]: The environment will be always testing?

File: soccer_application/config/config.py (hunk 6 above)
Comment (assaftal391) [reply soccer_application/config/config.py:15]: Sorry for the stupid question. When I'm printing "ConfigName.DEVELOPMENT" I get "ConfigName.DEVELOPMENT" back. When using value I get "dev" back (which what we want). What am I missing here?

**Review by YossiHananya — APPROVED** [review]
